src/lib/utils/string.js

```diff
@@ -12,9 +12,8 @@ export const truncateMiddle = (string, maxLength = null, ellipsis = '...') => {
     return string
   }
 
-  const halfLength = Math.floor(string.length / 2 - ellipsis.length)
-  const firstHalf = string.slice(0, halfLength)
-  const lastHalf = string.slice(-halfLength)
+  const firstHalf = string.slice(0, maxLength)
```

Comment:
```suggestion
  const halfLength = Math.floor(string.length / 2 - ellipsis.length)
  const firstHalf = string.slice(0, halfLength)
  const lastHalf = string.slice(-halfLength)
```

Reply:
I don't understand how it should work. The maxLength event not used in your implementation. It just split the string in a middle without shortening it.

Reply:
@yaroslav-fedyshyn-nordwhale sorry. my fault.
have updated the formula

1. we have 29 max length
2. subtracting ellipsis "..." length: 29 - 3 = 26
3. diving by 2, taking just whole part: [26 / 2] = [13,0] = 13
4. cutting first 13, last 13, composing with "..." onto a target truncated string